### ISSUE TITLE
Regression introduced few hours ago

### ISSUE THREAD

**Issue by peterdeme** [hadolint/hadolint#551]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

This bug was introduced in https://github.com/hadolint/hadolint/pull/548

### Expected behavior
This should work fine:

```dockerfile
ARG SERVICE_NAME

ENV SERVICE_NAME=${SERVICE_NAME}
```

### Actual behavior
Throws error:
```
/dev/stdin:17 DL3044 error: Do not refer to an environment variable within the same `ENV` statement where it is defined.
```

### Steps to reproduce the behavior

**Dockerfile** (if relevant)

```Dockerfile
FROM python:3.8-alpine

ENV SERVICE_NAME=$SERVICE_NAME

CMD ["/bin/bash"]
```

**Comment by lorenzo** [comment]
I will revert this and merge again after working the problems out

**Comment by lorenzo** [comment]
I have reverted it. cc @m-ildefons

**Comment by mvoitko** [comment]
This issue is still relevant for the hadolint:latest-debian Docker image

**Comment by lorenzo** [comment]
We just re-merged the feature with some fixes. Open a new issue with your case if something broke for you
